### ISSUE TITLE
Linux version?

### ISSUE THREAD

**Issue by robaca** [ForkIssues/Tracker#153]
Hi, 

I really would love to see Fork running on Linux. Are there any plans for this?

Best regards,
Carsten

**Comment by DanPristupov** [comment]
@robaca I've never developed for Linux and don't even know what languages and libraries people use to implement GUI applications in Linux. So I think this will not happen in the visible future.

**Comment by clounie** [comment]
An Electron app could be an option. Apps like [MongoDB Compass](https://www.mongodb.com/products/compass) use it.

That'd also get you Windows as a bonus.

**Comment by nebhale** [comment]
One of the reasons that I love Fork over all of the other options in this space is that it’s a native macOS application.  Electron is a reasonable choice for cross-platform applications, but there is no question when I use applications based on it, that they are not nearly as well integrated into _any_ operating system as applications built specifically for that operating system.  It seems that so far that @DanPristupov has had the goal of making the best Git GUI client on macOS and his focus on a single operating system has enabled him to do that without compromise.

**Comment by clounie** [comment]
To be clear, I'm not at all saying the macOS app should be converted to Electron.

I don't disagree - but I would rather *have* an app for Linux than not have one (which is what non-Electron equates to, probably)

Windows is already on the horizon (per the website,) and something like Electron is something that would need discussion **before** work is started on that. Hence my first comment.

**Comment by clounie** [comment]
Ended up switching to Linux at work, ironically - didn't know until a week ago.

Hope Fork does well.

**Comment by DanPristupov** [comment]
@clounie ouch, it's a big loss :(. You've been helping making Fork better from the day one (the initial post on HN).

I appreciate your help. Thank you very much ❤️.

**Comment by clounie** [comment]
![image](https://user-images.githubusercontent.com/11947182/45055385-df96b880-b04c-11e8-9755-457235e082b9.png)

If there's anything I could do to save Linux computers from GitKraken, let me know ;)

**Comment by maksimovic** [comment]
Slightly offtopic - I've tried GitKraken on Mac and it was literally unusable with our repository (it's huge). No person in the world could get used to waiting for 3-5s at least for _every click_. I also find Slack making my Mac going into airplane engine mode from time to time (being logged to multiple workspaces greatly "helps" in achieving that). So... I don't know, Electron may be okay sometimes, but for apps that should handle heavy lifting I don't think it's a good solution.

**Comment by clounie** [comment]
^ yep, agreed. The performance diff b/w Fork and GitKraken is pretty huge even on small repos.

**Comment by TheMrAnderson** [comment]
I would love Fork on Linux.  I'm using GitKraken at the moment

**Comment by coaperator** [comment]
I support!
I'm trying to leave Windows for Debian
Tired of endless glitches from Microsoft
We look forward to fork under Linux
Linux ❤️ Fork

**Comment by DanPristupov** [comment]
I'd not expect Fork on Linux until a decent UI dev environment appears there.

Avalonia looks very promising, but it must become mature.

**Comment by coaperator** [comment]
> I'd not expect Fork on Linux until a decent UI dev environment appears there.
> 
> Avalonia looks very promising, but it must become mature.

Yes, it looks pretty ..
But it is developing very slowly

**Comment by coaperator** [comment]
Well that's all, I switched from Windows to Debian ..
I really look forward to Fork under Linux, because, except for SmartGit, then there are simply no normal, adequate clients under Linux ..

**Comment by clounie** [comment]
> I'd not expect Fork on Linux until a decent UI dev environment appears there.
> 
> Avalonia looks very promising, but it must become mature.

QT?

**Comment by astrolemonade** [comment]
I would really love to see Fork available on Linux. It can be made to be built from source or you can distribute AppImages or snaps or flatpacks for cross distributions so you would not have to work with different package managers.
It could be made in Gtk/Qt/React Native/Flutter/wxWidgets.
I hope the best for Fork and I hope that it will come to Linux at some point !
Edit: Do not close this issue and do not make the same mistake Atlassian is doing, to ignore the Linux community. It is bigger than the Windows one in terms of how many people know to code!

Edit2: If you are wondering what Qt is capable of just look at Qbittorrent:https://github.com/qbittorrent/qBittorrent
It uses Qt/C++ and a library named libtorrent. Also Qt is cross platform and easy to write software with! There is a lot of information about it and there is also an implementation for Python if you are interested in the language.

**Comment by DanPristupov** [comment]
>  If you are wondering what Qt is capable of just look at Qbittorrent:https://github.com/qbittorrent/qBittorrent

![iu](https://user-images.githubusercontent.com/618115/65384766-697ff500-dd26-11e9-82b4-c81668a4a558.jpg)

This is kind of the reason, why I'd like to stay with the native libraries :)

**Comment by astrolemonade** [comment]
I don't quite understand what are you saying ! Do you like it or not ? If not, what are your reasons? (I would like to hear them for future projects) 
Gtk is the native for most of the linux distros out there

**Comment by TheMrAnderson** [comment]
I'm not sure what Fork is written in now, but Python with a front end like PyQT, Tkinter or WxPython looks promising, plus it's all cross platform.  I'm sure there are enough devs who would love to help in any way they could.

**Comment by neonb** [comment]
> This is kind of the reason, why I'd like to stay with the native libraries :)

For KDE and LXQt distros Qt _is the native library_ (e.g. Kubuntu, openSUSE, Lubuntu). A lot of Linux installations run some KDE/Qt software, regardless of desktop environment. They don't necessarily look like an outdated torrent client either – in fact, since a couple years back, the Plasma desktop environment is IMO among the best looking ones on the market.

If you'd like to go with the toolkit that's the closest to native to the largest portion of Linux userbase, it's GTK. (Avalonia bundles GTK, if you prefer to use it.) Qt and GTK both have pseudo-themes that look like each other, which both just delegate rendering to the underlying native toolkit. And I would describe both, especially Qt, a "decent" and "mature" UI dev environment. They've both got IDEs, XAML-style UI serialization, and 20+ years under the belt.

Tk and wx toolkits mentioned above are quite old, but can also leverage the DE's native toolkit with... varying success. All these should have licenses compatible with proprietary software, and bindings to any language you can think of.

**Comment by clounie** [comment]
^^ thank you for saying it better than I could have :)

**Comment by tflo** [comment]
I have nothing against a Linux version but please, @DanPristupov, keep the native Mac version. If you throw a Qt or Electron app at me, I’m out.

**Comment by joostoudeman** [comment]
Now, I'm not currently in a position to test. But wouldn't it be possible to run the windows version on linux using https://www.mono-project.com/? The windows version is based on .NET is it not?

**Comment by nartc** [comment]
@clounie (and any Linux users) have you looked at Sublime Merge? It's not Electron sh*t. and @DanPristupov I love Fork at the moment. I'm switching from Sublime Merge to Fork for maybe a sprint to see how it goes.

**Comment by clounie** [comment]
Hah you won't regret ;)

Just downloaded - interesting, will try using that instead of Kraken for a bit and see how it goes. Thanks :+1:

**Comment by Ansraer** [comment]
I am using Fork on a Win PC I have at home, but have Linux on my laptop and work pc. I would really love a port, have to use gitKraken right now and it sucks.
I know that this would be a massive undertaking, but based on what I have heard about net core and Avalon it should at least be possible without rewriting everything. 

If Linux support is ever considered know that I am willing to help in any way I can.

**Comment by MisterJD** [comment]
Yes a additional Linux version would be fantastic. 👍

**Comment by dmansurov83** [comment]
@DanPristupov Hello! Which stack you are using for the app? Is it related to .net?

**Comment by dmansurov83** [comment]
`Fork-1.45.0-full.nupkg` OK. Obviously this is .net) As I know Mono is avalible for linux. Another promising solution may be https://github.com/AvaloniaUI/Avalonia

**Comment by saisurya-kat** [comment]
@DanPristupov Actually Gtk+ is a very good option, in future if you decide to support Linux.
elementary OS focuses on Native app development with a **dedicated SDK** for developers with "**Pay what you want**" app store, Gtk+ apps can be developed in **Vala** (an object oriented language which is **very similar to C#**
For more details:  [Developer Elementary](https://developer.elementary.io/). 
![image](https://user-images.githubusercontent.com/8770193/76701496-442bdc80-66f4-11ea-8f9d-9e305fd32419.png)
![image](https://user-images.githubusercontent.com/8770193/76701504-56a61600-66f4-11ea-9481-c4e1c4e6e295.png)
These are native Linux apps (171+) curated for elementary OS https://appcenter.elementary.io/
![screencapture-appcenter-elementary-io-2020-03-15-19_39_27](https://user-images.githubusercontent.com/8770193/76701545-bc929d80-66f4-11ea-9979-53c31a01eab4.png)

**Comment by Haprog** [comment]
I think I used Fork for about 1-2 years on macOS but I just recently switched from macOS to Windows (for about 6 months) and then to Linux (Ubuntu based) and sadly I can't use Fork anymore because it's not available on Linux.

Currently I'm using the free version of GitKraken but I'm considering paying for GitKraken (as an individual) or might get my employer to pay for it, but I would much rather buy Fork if only it was available on Linux too. (Though I think I might be going back to macOS in the future, depending on what MacBook Pro hardware is like at the time, and will probably reconsider using Fork again then).

I haven't noticed any bad performance with GitKraken so far (been using it for several months now with decent size repos), but maybe it's just because of my high end hardware.

There are obviously many different ways how to approach developing a Linux app or a cross-platform app. I know Electron is popular and should be a low-effort way to be cross platform. Though now that you already have native Win and macOS apps it's obviously a different situation than developing a new app. I know many Electron apps have some performance issues though I'm not sure if it's because of Electron itself or because they just haven't been well optimized for the Electron stack.

Even if using Electron I think you can probably get good performance if you do it right utilizing web workers where it makes sense and optimizing for browser rendering engine. For some specific tasks WebAssembly could also be useful (used within Electron).

Another cross platform approach would be a Java app. Good examples of very complex but surprisingly performant apps are all the [IDEs by JetBrains](https://www.jetbrains.com/products.html#type=ide) e.g. IntelliJ IDEA, PHPStorm, WebStorm, PyCharm, Rider, CLion etc. (afaik those are made with Java). Source code for the IntelliJ IDEA Community Edition is available here https://github.com/JetBrains/intellij-community (apparently it's basically a Java Swing application)

**Comment by pr8x** [comment]
In case @DanPristupov decides to tackle Linux support at some point, [Avalonia](https://github.com/AvaloniaUI/Avalonia) would be a great way to start. Very similar to WPF, in fact, large parts of the Windows/WPF implementation could be ported (I suppose).

**Comment by limbusdev** [comment]
You can write it in C#, for GUI you'll actually need something like GTK or Qt. It would be soooo great to have Fork for Linux.

**Comment by limbusdev** [comment]
I'd be one of the first to purchase it : ) Since .NET Core is available for linux, this could be a chance: https://github.com/AvaloniaUI/Avalonia

**Comment by moqmar** [comment]
Any update on this? To this date, GitKraken is still the only comparable Git client for Linux, the Linux world could really use some great native alternative. :upside_down_face:

**Comment by epsimatic** [comment]
> GitKraken is still the only comparable Git client for Linux

Try [Sublime Merge](https://www.sublimemerge.com) for Linux. It doesn't look native, but still a good solution for Linux, and it's miles ahead of browser apps. 

<img width="1207" alt="sshot 2021-03-24 в 14 28 37" src="https://user-images.githubusercontent.com/1821846/112286694-649a5980-8cad-11eb-8045-7ab062e9a8dc.png">

**Comment by noskap** [comment]
FWIW [Flutter 2 just released with native linux support](https://flutter.dev/desktop) 
I personally haven't tried it (yet)

**Comment by HighPriest** [comment]
Just stumbled onto this:
https://github.com/dotnet/maui

What do you think @DanPristupov

**Comment by DanPristupov** [comment]
@HighPriest
Just check the official press-release: https://devblogs.microsoft.com/visualstudio/visual-studio-2022/

Here is the image they use to advertise a **multi-platform** UI library.

1. 'Click me' button. Is a center-aligned hyperlink on iOS and Mac, however it's is a left-aligned button on Windows and is a full-width button on Android (with both left and center alignment).
2. Checkbox is left-aligned on iOS and right-aligned on Android.
3. The 'you clicked' label is center aligned on some devices and left-aligned on other ones.

I would never touch a UI library from developers with such an attention to details. MAUI is only being taken somehow seriously because it's made by MS.

![maui](https://user-images.githubusercontent.com/618115/115543408-9f3de480-a2a1-11eb-9042-27f012ba76ab.png)

**Comment by gbtb** [comment]
I've managed to run Fork under Wine, and it is more or less usable. It seems that nowadays mono compiler has a very decent support for WPF. Considering UI, Fork runs great, even though without hardware acceleration. There are some problems with bundled git (which sadly cannot be overridden to native one) and credentials manager, but luckily they are not hard blockers and can be bypassed.
Test report with required workarounds is at [winehq.org](https://appdb.winehq.org/objectManager.php?sClass=version&iId=39135)

**Comment by starry-abyss** [comment]
So now it's mostly a matter of the author to make a Linux build and no extra GUI libraries are needed anymore?

**Comment by pagdot** [comment]
> So now it's mostly a matter of the author to make a Linux build and no extra GUI libraries are needed anymore?

Not quite. As far as I understand, the Windows  version of Fork is developed with the GUI framework WPF. The *multi-platform UI* framework mentioned above is *MAUI*, the successor to *Xamarin Forms*. This is a different UI framework. So it is not just "adding a new build", but creating the UI on top of a new framework. Much of the code base may be reused (except UI specific parts).

Ad advantage would be that they could use one code base to target the existing platforms (Mac, Windows), instead of having (I'm just guessing here) very different applications for Windows and Mac, although they look and feel the same.

**Comment by starry-abyss** [comment]
@pagdot So this statement above my comment is not true?

> It seems that nowadays mono compiler has a very decent support for WPF. Considering UI, Fork runs great, even though without hardware acceleration.

**Comment by pagdot** [comment]
Sorry somehow skipped the comment in my mind :see_no_evil:

**Comment by cfazilleau** [comment]
Just noticed that Fork has been sponsoring [Avalonia](https://github.com/AvaloniaUI/Avalonia) since January 2022 
![image](https://user-images.githubusercontent.com/35767293/165534117-049fe72e-cedb-4fdb-99c0-0fe5a9a3bcfe.png)
Does it mean you are switching fork to Avalonia for the linux version ?

**Comment by zrpfpr** [comment]
One step closer, maybe? :)

https://ubuntu.com//blog/install-dotnet-on-ubuntu

**Comment by pagdot** [comment]
dotnet core is available on linux for some time already. THe issue is that WPF, Xamarin, WinForms aren't available

**Comment by f03n1x** [comment]
Just adding a comment here to just say that I'd love Fork on Linux. Installed Linux to test my game on a Linux build and sadly I'm going to have to find an alternative to Fork :( for the time being at least!

**Comment by Sergio-Dobrianskiy** [comment]
Same here, it would be really nice to have Fork available on Linux.
@f03n1x have you found a decent alternative?

**Comment by f03n1x** [comment]
> Same here, it would be really nice to have Fork available on Linux. @f03n1x have you found a decent alternative?

I kinda just settled with github for desktop for the time being, but eh not a fan lol, I know gitkraken is a thing but that's what I used prior to Fork, my issue with that was having a repo with 2000 + commits would cause mega lag

**Comment by MacGurk** [comment]
Would also love to use Fork on Linux. Used it for quite some time in Widows and now switched to Fedora for work. For now I test out Git Kraken, but already missing some features.

**Comment by Lovasz-Akos** [comment]
![2ombPrayGIF](https://github.com/fork-dev/Tracker/assets/55391231/cfe24b39-1d69-489f-805c-264193a35c29)
do it

**Comment by iamajoe** [comment]
I work on Mac and on Linux and I was reluctant to use Fork to be honest. I didn't like to leave the terminal, I didn't like GitKraken... A colleague converted me and now, when I am on Linux, I miss Fork so much. To a point that I go to my Mac to handle the git hygiene and come back to Linux afterwards. I would gladly pay for it. Just one more push ahead for the port.

**Comment by gamedevsam** [comment]
For what it's worth, I would also buy GitFork on Linux (<3 ZorinOS), but understand the likely hood of that happening is low. SublimeMerge appears to be a reasonable alternative, not as feature filled, and double the price, but it's that or GitKraken for Linux users unfortunately.

**Comment by HarrySpyrou** [comment]
I'm adding here, I'm waiting for the Linux version to buy gitfork. I'm using Windows at work, but 90% of my coding time goes on Linux aka not Fork.

**Comment by Jademalo** [comment]
Guess I'll add myself to the list, do most of my development on Linux so it would be hugely beneficial to have

**Comment by EmanueleSpadaro** [comment]
Adding myself to the growing list, I can't find myself using any client besides Fork

**Comment by tan-wei** [comment]
Hmm. If Linux version is released, I will buy fork ASAP, because I truly want to use it on all the platform, and mostly I use Linux.

**Comment by f03n1x** [comment]
> Hmm. If Linux version is released, I will buy fork ASAP, because I truly want to use it on all the platform, and mostly I use Linux.

Same here! I'll buy another license copy to show my support for it. Is there maybe some way we could look into having the linux version open sourced? if it's not hard to do/if the original developer cannot work on it, would we be able to voluntarily work towards making a linux version?

**Comment by LeVuMinhHuy** [comment]
any progress on this guys ?

**Comment by joepbuhre** [comment]
Wanting this as well, would also buy a license if that helps 🙏🏻

**Comment by NitroHxC** [comment]
I switched to Linux completely earlier this year and spent WEEKS trying to make Fork work with Wine but apart from opening the GUI, I got many issues with the bundled git (and especially with ssh, in fact I couldn't clone anything) as @gbtb mentioned.

So far I've been using Fork inside a Windows VM (qemu) in which I also store my repository as a Samba drive. I open the project in any IDE from my Linux host and the editing works ok.

It adds a non-negligible overhead when compiling... but at least I can use Fork!

@DanPristupov If a full Linux port is out of discussion, what about trying to find a workaround for Wine in handling the git and ssh subprocesses? 
As many other mentioned, I'd be happy to pay a plus and even an annual subscription for it, as honestly I don't see any good alternative in any existing git client.

**Comment by Griefed** [comment]
Another brick in the wall:

I've come to love Fork, after using SourceTree for quite some time. I've tried GitKraken, GitHub Desktop,even git CLI and more, but none of these were as satisfying as Fork has been. Fork convinced me in a matter of minutes and I've thrown money at it quicker that I could say NullPointerException.

Due to the recent changes in Microsoft's service agreement, I've take a serious look at switching to Linux. So far, so good. Sure, there are drawbacks when you're used to the level of convenience Windows brings, but on the plus side, I'd have a Linux under my arse.
Buuuuuuut I can't get Fork to run in Linux and...that's probably going to be the reason why I will move back to Windows. Good git GUIs are few and far between and Fork is a gods-send. Would hate to miss it.

**Comment by arthurka-o** [comment]
Finally freed up from the Apple shackles, and now the most missing app from my work stack is Fork.
I want to thank the devs for building one of the most useful and used applications. From university to startups, I could always rely on Fork, its magnificent rebase functionality, and the commit tree.

Would buy it again for the Linux support in a heartbeat! And thanks for the one-time license purchase in an age where even car seats have subscriptions.

**Comment by gamedevsam** [comment]
I want to give the Linux crew a bit of an update. I've been using the Windows version of GitFork with WSL (Windows Subsystem for Linux) and it's been surprisingly stable and reliable. I know for some it's a matter of principle, but if you don't care about using Windows, Fork + WSL works great!

**Comment by f03n1x** [comment]
> I want to give the Linux crew a bit of an update. I've been using the Windows version of GitFork with WSL (Windows Subsystem for Linux) and it's been surprisingly stable and reliable. I know for some it's a matter of principle, but if you don't care about using Windows, Fork + WSL works great!

That sounds like a good alternative (for the time being, I do hope linux build is coming though!), I had some very painful issues with setting up a git repo on my linux pc, so much so I had to just cave and install windows alongside it, basically a bunch of ssh issues that I couldn't be bothered spending more than 4+ hours looking into fixing (I did spend quite a long time trying to fix it but I needed to setup the repo for work also found out how annoying github for desktop is, in where it does nothing to help with the ssh stuff).

**Comment by Baardi** [comment]
This is gitg, made using the gtk framework. I find it visually pleasing, and not too different looking than Git Fork:

![Skjermbilde fra 2023-11-11 20-32-54](https://github.com/fork-dev/Tracker/assets/19634818/03c1b48c-5095-48ea-b51d-49dd29bd7ee8)

The problem about gitg is that it is lacking many basic features, configurability and updates. Git Fork for Linux would've been fantastic

**Comment by f03n1x** [comment]
Looks really good! Very similar to gitfork indeed

**Comment by scriptsengineer** [comment]
> This is gitg, made using the gtk framework. I find it visually pleasing, and not too different looking than Git Fork:
> 
> ![Skjermbilde fra 2023-11-11 20-32-54](https://user-images.githubusercontent.com/19634818/282262083-03c1b48c-5095-48ea-b51d-49dd29bd7ee8.png)
> 
> The problem about gitg is that it is lacking many basic features, configurability and updates. Git Fork for Linux would've been fantastic

It simply crashes with repositories with large numbers of commits

**Comment by jonaaix** [comment]
@DanPristupov In theory if we'd create an open source app from scratch, that has as a similar UI layout as fork, would that be acceptable?

**Comment by HarrySpyrou** [comment]
> @robaca Hi. I've never developed for Linux and don't even know what languages and libraries people use to implement GUI applications in Linux. So I think this will not happen in the visible future.

It's been 6 years.

**Comment by Miguel33Angel** [comment]
> This is gitg, made using the gtk framework. I find it visually pleasing, and not too different looking than Git Fork:
> 
> ![Skjermbilde fra 2023-11-11 20-32-54](https://user-images.githubusercontent.com/19634818/282262083-03c1b48c-5095-48ea-b51d-49dd29bd7ee8.png)
> 
> The problem about gitg is that it is lacking many basic features, configurability and updates. Git Fork for Linux would've been fantastic

It's open source so anyone could improve it, but I've been having a hard time actually ccompiling it from source code...

**Comment by ruben-grossmann** [comment]
So do we just leave a comment to upvote this topic? If so, done!

**Comment by andgeno** [comment]
Would love to see a port to Linux. I'd also help porting it if needed.

**Comment by julaudo** [comment]
Currently fork is working well with Wine stable (wine-8.0.2) under Debian 22.04
I have a basic usage (mainly rebases and conflict resolution) and used it without any issue so far.

Note : I'm using SSH authentication, it's working correctly

**Comment by EvHaus** [comment]
@julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue (https://github.com/bottlesdevs/Bottles/issues/202)

**Comment by julaudo** [comment]
> @julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue ([bottlesdevs/Bottles#202](https://github.com/bottlesdevs/Bottles/issues/202))

Actually I did not run the installation exe (because of the bug you mentionned). Instead of that I followed this steps:
- Download **Fork-1.91.2.exe** for fork website
- Extract files contained in **Fork-1.91.2.exe**
- One of those files is named **Fork-1.91.2-full.nupkg**, extract it
- Fork executable is located at **Fork-1.91.2-full/lib/net45/Fork.exe**

I just had to install dotnet462 (using winetricks) and it ran fine.

**Comment by DanPristupov** [comment]
The latest (1.92.0) .nupkg file can be downloaded here: https://fork.dev/update/win/Fork-1.92.0-full.nupkg

**Comment by vuvarov-cantaloupe** [comment]
It seems that Fork is written in .NET. There are couple modern options to write native or native looking UI for all 3 major platforms:
* [Uno Platform](https://platform.uno/#)
* [Avalonia](https://avaloniaui.net/)
* [MAUI fork](https://github.com/jsuarezruiz/maui-linux) with Linux support

If Fork is written in WPF it shouldn't be to hard to port it to another UI framework.

**Comment by gbtb** [comment]
> It seems that Fork is written in .NET. There are couple modern options to write native or native looking UI for all 3 major platforms:
> 
>     * [Uno Platform](https://platform.uno/#)
> 
>     * [Avalonia](https://avaloniaui.net/)
> 
>     * [MAUI fork](https://github.com/jsuarezruiz/maui-linux) with Linux support
> 
> 
> If Fork is written in WPF it shouldn't be to hard to port it to another UI framework.

For the sake of completeness, it's also worth mentioning that Avalonia recently announced their own cross-platform WPF implementation called [XPF](https://avaloniaui.net/XPF). It's paid solution aimed mostly to big enterprises, as I see it. But maybe better conditions can be negotiated with them.

**Comment by andgeno** [comment]
> > @julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue ([bottlesdevs/Bottles#202](https://github.com/bottlesdevs/Bottles/issues/202))
> 
> Actually I did not run the installation exe (because of the bug you mentionned). Instead of that I followed this steps:
> 
> * Download **Fork-1.91.2.exe** for fork website
> * Extract files contained in **Fork-1.91.2.exe**
> * One of those files is named **Fork-1.91.2-full.nupkg**, extract it
> * Fork executable is located at **Fork-1.91.2-full/lib/net45/Fork.exe**
> 
> I just had to install dotnet462 (using winetricks) and it ran fine.

I ran `winetricks dotnet462`. Then, I tried to launch Fork with `wine ./lib/net45/Fork.exe` and end up with this dialog.

![image](https://github.com/fork-dev/Tracker/assets/2432608/e5f9ed2b-f254-42b0-8094-be17b4072cff)

What am I doing wrong/differently?

**Comment by julaudo** [comment]
> > > @julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue ([bottlesdevs/Bottles#202](https://github.com/bottlesdevs/Bottles/issues/202))
> > 
> > 
> > Actually I did not run the installation exe (because of the bug you mentionned). Instead of that I followed this steps:
> > 
> > * Download **Fork-1.91.2.exe** for fork website
> > * Extract files contained in **Fork-1.91.2.exe**
> > * One of those files is named **Fork-1.91.2-full.nupkg**, extract it
> > * Fork executable is located at **Fork-1.91.2-full/lib/net45/Fork.exe**
> > 
> > I just had to install dotnet462 (using winetricks) and it ran fine.
> 
> I ran `winetricks dotnet462`. Then, I tried to launch Fork with `wine ./lib/net45/Fork.exe` and end up with this dialog.
> 
> ![image](https://private-user-images.githubusercontent.com/2432608/289289047-e5f9ed2b-f254-42b0-8094-be17b4072cff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODk1OTYsIm5iZiI6MTcwMjI4OTI5NiwicGF0aCI6Ii8yNDMyNjA4LzI4OTI4OTA0Ny1lNWY5ZWQyYi1mMjU0LTQyYjAtODA5NC1iZTE3YjQwNzJjZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTFUMTAwODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQyNzExYTQ4OTBiNzg2MDE2OGMyODFlZmZmNTExOTg0NGM4MTk5NzU5YmU3ZmQwNzdmYjg5YWMwM2UxOWZiMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RGwC0qNaHb1t68UJEUpOcOFGmZbCcXyqzlCB3_uZ2EQ)
> 
> What am I doing wrong/differently?

Clicking on yes should do the job

**Comment by andgeno** [comment]
> > > > @julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue ([bottlesdevs/Bottles#202](https://github.com/bottlesdevs/Bottles/issues/202))
> > > 
> > > 
> > > Actually I did not run the installation exe (because of the bug you mentionned). Instead of that I followed this steps:
> > > 
> > > * Download **Fork-1.91.2.exe** for fork website
> > > * Extract files contained in **Fork-1.91.2.exe**
> > > * One of those files is named **Fork-1.91.2-full.nupkg**, extract it
> > > * Fork executable is located at **Fork-1.91.2-full/lib/net45/Fork.exe**
> > > 
> > > I just had to install dotnet462 (using winetricks) and it ran fine.
> > 
> > 
> > I ran `winetricks dotnet462`. Then, I tried to launch Fork with `wine ./lib/net45/Fork.exe` and end up with this dialog.
> > ![image](https://private-user-images.githubusercontent.com/2432608/289289047-e5f9ed2b-f254-42b0-8094-be17b4072cff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODk1OTYsIm5iZiI6MTcwMjI4OTI5NiwicGF0aCI6Ii8yNDMyNjA4LzI4OTI4OTA0Ny1lNWY5ZWQyYi1mMjU0LTQyYjAtODA5NC1iZTE3YjQwNzJjZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTFUMTAwODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQyNzExYTQ4OTBiNzg2MDE2OGMyODFlZmZmNTExOTg0NGM4MTk5NzU5YmU3ZmQwNzdmYjg5YWMwM2UxOWZiMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RGwC0qNaHb1t68UJEUpOcOFGmZbCcXyqzlCB3_uZ2EQ)
> > What am I doing wrong/differently?
> 
> Clicking on yes should do the job

Sorry, I wasn't clear about what would happen after clicking the 'yes' button.

Unfortunately, it doesn't run Fork but opens a browser with this URL:

https://dotnet.microsoft.com/en-us/download/dotnet-framework/net481?cid=getdotnetframework

There is no other error message.

I'm using Ubuntu 22.04 LTS, fresh install.

EDIT:

Actually, there are errors when I run `winetricks dotnet462` that might be the issue here:

```
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing load_dotnet45 
------------------------------------------------------
warning: This package (dotnet45) is broken in wine-6.0.3. Broken since 5.18. See https://bugs.winehq.org/show_bug.cgi?id=49897 for more info. Use --force to try anyway.
------------------------------------------------------
```

So, I guess my installation of Wine or Winetricks is too old?

**Comment by Baardi** [comment]
> > > > @julaudo Can you share the installation steps you used? When I tried it I got blocked by this issue ([bottlesdevs/Bottles#202](https://github.com/bottlesdevs/Bottles/issues/202))
> > > 
> > > 
> > > Actually I did not run the installation exe (because of the bug you mentionned). Instead of that I followed this steps:
> > > 
> > > * Download **Fork-1.91.2.exe** for fork website
> > > * Extract files contained in **Fork-1.91.2.exe**
> > > * One of those files is named **Fork-1.91.2-full.nupkg**, extract it
> > > * Fork executable is located at **Fork-1.91.2-full/lib/net45/Fork.exe**
> > > 
> > > I just had to install dotnet462 (using winetricks) and it ran fine.
> > 
> > 
> > I ran `winetricks dotnet462`. Then, I tried to launch Fork with `wine ./lib/net45/Fork.exe` and end up with this dialog.
> > ![image](https://private-user-images.githubusercontent.com/2432608/289289047-e5f9ed2b-f254-42b0-8094-be17b4072cff.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIyODk1OTYsIm5iZiI6MTcwMjI4OTI5NiwicGF0aCI6Ii8yNDMyNjA4LzI4OTI4OTA0Ny1lNWY5ZWQyYi1mMjU0LTQyYjAtODA5NC1iZTE3YjQwNzJjZmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMTFUMTAwODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWQyNzExYTQ4OTBiNzg2MDE2OGMyODFlZmZmNTExOTg0NGM4MTk5NzU5YmU3ZmQwNzdmYjg5YWMwM2UxOWZiMSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.RGwC0qNaHb1t68UJEUpOcOFGmZbCcXyqzlCB3_uZ2EQ)
> > What am I doing wrong/differently?
> 
> Clicking on yes should do the job

I didn't get the same problem. I was able to install .NET Framework 4.6.2 without a problem. But as I attempt to actually start Fork it immediately crashes, on Fedora 39. Using the same install method as you @julaudo. I can send the crashlogs, if it is of any interest to @DanPristupov

**Comment by DanPristupov** [comment]
@Baardi send them to support@fork.dev

**Comment by Baardi** [comment]
> @Baardi send them to [support@fork.dev](mailto:support@fork.dev)

Thanks for the help. He was able to figure out what was wrong in my case:

At first I wasn't able to launch due to missing fonts. 
- I ran "sh winetricks corefonts" which solved that issue.

Now I could get to the welcome screen. But I immediately got the following error afterwards;
![Screenshot from 2023-12-12 22-38-25](https://github.com/fork-dev/Tracker/assets/19634818/47d22f52-5a40-4cc7-89fa-41566ead7f22)

- This time, the problem was that I had unzipped Fork to the wrong location. The content of lib/net45 has to be unzipped to %localappdata%/Fork/app-%version%
The content of gitInstance should go to %localappdata%/Fork/gitInstance. Then it worked for me.

**Comment by DanPristupov** [comment]
@Baardi so, does it really work on Linux? 👀

May be we can create a tutorial (or even an .sh script) which would install Fork.

**Comment by zrpfpr** [comment]
Great work @Baardi!
In that case, you need windows version for external Diff/Merge tools also, right? Or you can bypass at some point and use your Linux programs?

**Comment by Baardi** [comment]
I think I was a bit too optimistic. The initial results seemed good.
I was able to open a local git repository, see file changes, etc, but when I dragged it into a different monitor, other than the main one, it immediately froze. It would also launch next time in the same broken state. The only way I found to fix it was to delete the settings.json file.
![Screenshot from 2023-12-13 18-33-35](https://github.com/fork-dev/Tracker/assets/19634818/11e491c1-0c34-4cb1-b822-6ecc7d932fe6)

Next I tried creating a new branch. sh.exe crashed. But the branch was succesfully created.
![Screenshot from 2023-12-13 18-32-38](https://github.com/fork-dev/Tracker/assets/19634818/13d57a98-d20d-4904-9851-4074e53f9ec7)

I then tried commit a change to that branch. That worked fine, no errors.
I then tried pushing the changes. It didn't work, as libcurl-4.dll was missing.
![Screenshot from 2023-12-13 18-24-39](https://github.com/fork-dev/Tracker/assets/19634818/443f8085-2fc5-4269-b70f-14518dbc383f)

So it's safe to say I have a few issues to sort out.


EDIT: So I don't spam you all. The dll load failure was due to a missing dll in system32/syswow64: Wldap32.dll. I copied the dlls from my windows 11 installation, which seemed to work. 
Authentication is still a problem though, at least for github (Neither https or ssh worked for me).

**Comment by Baardi** [comment]
I think I probably just got as far as @gbtb and @NitroHxC. Curious whether or not you got further than that @julaudo

**Comment by DanPristupov** [comment]
To discuss running Fork with Wine I created a separated issue : https://github.com/fork-dev/Tracker/issues/2033

**Comment by theburntcrumpet** [comment]
I just wanted to drop a comment and say that Fork on Linux is definitely something there's some demand for. I've yet to find a git UI on linux as clean as fork which handles hooks/git-lfs properly and would love to see it natively supported - I'd happily buy another license just for linux. Really appreciate fork, keep up the good work.

**Comment by lucastucious** [comment]
I'm here just for saying that fork is my best friend. But I'm moving to linux. I hope I will see my friend one day

**Comment by daltonbr** [comment]
I want to express my support here - I am moving out from GitKraken - which served me well, but I grew tired of the slowness of an Electron app and the subscription model - the native approach of Fork got me.

I use all Mac/Windows/Linux and I would love to see a native Linux release - would gladly pay a bit more for that privilege. 🐧

**Comment by mateusz-spychala** [comment]
I agree. We need good GIT client for linux

**Comment by safield** [comment]
I wonder if a reasonable option would be to just have an officially supported flatpak that bundles the windows version with wine and anything else needed to run. This way the devs could couple the windows version with a tested and known working version of wine and its wine configuration. I imagine through this technique you could probably provide a reliable experience.

**Comment by Feyko** [comment]
Fork is so good I'm making everyone at work switch to it instead of SourceTree but I sadly can't use it myself as I work on Linux :pensive: Please consider supporting linux!

**Comment by limbusdev** [comment]
> I wonder if a reasonable option would be to just have an officially supported flatpak that bundles the windows version with wine and anything else needed to run. This way the devs could couple the windows version with a tested and known working version of wine and its wine configuration. I imagine through this technique you could probably provide a reliable experience.

But you would want to use a native git. Not one running through wine.

**Comment by safield** [comment]
> > I wonder if a reasonable option would be to just have an officially supported flatpak that bundles the windows version with wine and anything else needed to run. This way the devs could couple the windows version with a tested and known working version of wine and its wine configuration. I imagine through this technique you could probably provide a reliable experience.
> 
> But you would want to use a native git. Not one running through wine.

Not necessarily. You would probably bundle git in the flatpak. That is the idealogy of flatpak....you bundle your dependencies. This is exactly how bundling in flatpak with a snapshot of all the dependencies  (including wine) you can provide a reliable and stable experience.

**Comment by sharpjs** [comment]
Fork is so good on my Windows workstation that it makes me reconsider running Linux on a new laptop I got.  Yes, gitg is similar, but it is _far_ from a complete substitute for Fork.  Fork's feature set and UX is just in a different league.

I've already converted one coworker to Fork, and a second one is just about there.

**Comment by ZoserLock** [comment]
I hope a version for linux arrive some day. I would gladly pay for that version again even if I already have the windows version.

**Comment by CzB404** [comment]
![GIF of Scruffy from Futurama saying "Seconded"](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExNTFncm44NjB3ODBlYWR4YjV2Mmt3M2FhODhoZ25yeDF2ZjRyeXdjaiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/7LAqMVFxOGPAc/giphy.gif)

**Comment by dtap001** [comment]
please make my dream came true and migrate git-fork to linux!! <3

**Comment by novadragonDOTspace** [comment]
Best thing for fork would be to be open source
Second best thing would be to just release it on linux

**Comment by safield** [comment]
> Best thing for fork would be to be open source Second best thing would be to just release it on linux

What?! Why??? The two developers have created an amazing product and a nice little business for themselves, and you want them to open source it?!? I vehemently disagree.

**Comment by novadragonDOTspace** [comment]
Open Sourcing an app doesnt exclude Monetisation. 

[aseprite/asesprite](https://github.com/aseprite/aseprite) would be a prime example of an paid open-source app, where you could either buy the built product.
Red Hat Enterprise Linux would also be another example.

There's also still several donation options. Sometimes you can even get a grant via that.

**Comment by BrodyB** [comment]
Want to also say it would be amazing to see Fork in the Linux ecosystem. I'm migrating myself over to Pop!_OS because (nervously looks at tech company trends) and this is the one big thing I'm missing.

The landscape of git clients on Linux is surprisingly bad! All the available ones are underbaked or unreliable. There is definitely a real need for a quality application that is not being met.

**Comment by ZaLiTHkA** [comment]
> The landscape of git clients on Linux is surprisingly bad! All the available ones are underbaked or unreliable. There is definitely a real need for a quality application that is not being met.

this is scarily true.. I recently moved my dev environment to Zorin OS, I've spent many hours over the past week trying to find a Git GUI that "just works". the biggest issue for me, is that virtually none of them support showing diffs between two selected commits.

in the meantime, I've switched over to Sublime Merge. if that had a perpetual license option, I would have made the move permanent already, but I'm holding off for a while to see what happens with Fork first. it would _really_ suck having to pay $99 (currently about R1800 in my local currency) every 3 years just to get realiable access to basic Git features in a GUI.

**Comment by VirtualOverrider** [comment]
+1 for native Linux support!
Fork could very easily eat the subscription options' lunch if only it were available. <3

**Comment by HermesSantos** [comment]
Please, I need it!

**Comment by tobysroof** [comment]
I've left windows and Fork is what I miss the most. It would be amazing to get support for Linux

**Comment by itcotta** [comment]
Been a long time Fork user and a Linux version would be huge!

**Comment by pfeerick** [comment]
Same here... I started using it on Windows on recommendation from a colleague who uses it on Mac, and haven't looked back. I've moved to Linux more for development, but reguarly switch between that and Windows, so would love to see Fork there also. Instead I've been using GitLens in VSCode, and plan to have a look at https://github.com/sourcegit-scm/sourcegit next while I wait for https://github.com/relagit/relagit to mature, as it looks to be a potential crossplatform option also.

**Comment by rafaribe** [comment]
Been using it in MacOS for the longest time and on Windows also. But recently switched to Arch Linux because of the Windows Recall privacy concerns and it's a shame that I have to use GitKraken or Gitlens. Please, if you can, support a Linux version

**Comment by theburntcrumpet** [comment]
Sourcegit https://github.com/sourcegit-scm/sourcegit is pretty nice for those still hoping for a linux version of fork

**Comment by dix** [comment]
Hi,

Another vote for a Linux version of the best git client ❤️

**Comment by gohryt** [comment]
And another.

**Comment by arnedeutscher** [comment]
And another!

**Comment by cesxhin** [comment]
And another!

**Comment by Tymisko** [comment]
And ANOTHER!!!! 🐧🐧🐧🐧🐧🐧

**Comment by kimmellj** [comment]
And another!

**Comment by OverHome** [comment]
> Sourcegit https://github.com/sourcegit-scm/sourcegit is pretty nice for those still hoping for a linux version of fork

Yes, Sourcegit is the best alternative!
I'm impressed by how good it is compared to all the others (Although maybe it's just my habit of Fork-like gitui).

**Comment by ZaLiTHkA** [comment]
> Yes, Sourcegit is the best alternative! I'm impressed by how good it is compared to all the others (Although maybe it's just my habit of Fork-like gitui).

no, it's not just you.. out of all the options I've tried so far, SourceGit is definitely the closest thing to a "Fork experience" on Linux.

with that said, it is still undergoing massive changes with almost weekly updates at the moment.. each of which seem to be adding a few more equivalent features that we're already used to having from Fork. for example: "chunk staging" was only added a few versions back and the buttons to do so (without using the context menu) were only added to this latest release the other day.

so Fork on Windows and SourceGit on Linux seems to be the way to go... at least for now.

**Comment by julaudo** [comment]
> > Sourcegit https://github.com/sourcegit-scm/sourcegit is pretty nice for those still hoping for a linux version of fork
> 
> Yes, Sourcegit is the best alternative! I'm impressed by how good it is compared to all the others (Although maybe it's just my habit of Fork-like gitui).

I just tried SourceGit and it seems very good! I think it is clearly inspired by fork. I just don't understand why it uses a monospace font almost everywhere in the UI. On this point, I prefer the fork UI, which uses a regular font.

**Comment by pfeerick** [comment]
You can configure what font it uses in the preferences. The author is also very responsive to questions and requests... i.e. I asked a question about how to change the submodule update command uses in SourceGit, and an hour later he added options for doing exactly that.

**Comment by worldofgeese** [comment]
The feature that would get me to use SourceGit on Linux is lean branching. It's my favorite feature in Fork.

**Comment by julaudo** [comment]
> You can configure what font it uses in the preferences. The author is also very responsive to questions and requests... i.e. I asked a question about how to change the submodule update command uses in SourceGit, and an hour later he added options for doing exactly that.

You're right, the author is very responsive. I raised several bugs/feature requests and most of them were closed within the day. I'm going to switch to SourceGit, thanks.

**Comment by tomvoros** [comment]
Another vote for Linux support! I'm on Windows but considering a switch to Linux. 😀

However, I appreciate the author's pickiness on a UI framework. Some of the common multiplatform frameworks like Qt just don't _feel_ good for some reason, and Fork has a highly polished UI.

**Comment by rzeidler** [comment]
And another!

**Comment by josvaal** [comment]
Como no hay ningùn hispanohablante, vengo a comentar xd, y es muy necesitado la versión para linux.

**Comment by Tiikoni** [comment]
I'd also love to have Linux version! I could move from Windows to Linux overnight without much noticing, as every program I use have also Linux version. Except Fork.

**Comment by rdmrtn** [comment]
> Sourcegit https://github.com/sourcegit-scm/sourcegit is pretty nice for those still hoping for a linux version of fork

sourcegit might be the best alternative for who use Fork in other platforms.

**Comment by mterzic** [comment]
One more here!

**Comment by mocakbarmaulana** [comment]
and another!

**Comment by ghost** [comment]
+Linux build.
Using windows version at work, but want it on the penguin for home

**Comment by PabloRascazzi** [comment]
And another!

**Comment by brycewilkins** [comment]
+1 for Fork on Linux! :pray:

**Comment by T4mmi** [comment]
+1 ... still can't wrap my mind around git-diffs and messy rebases on VSCode without the help of Fork

_dunno what frame/language you are currently using for Fork, but as cross-platform dev I mostly used Qt for GUIs ... and it works like a charm, without the performance issues of Electron_

**Comment by VictorCarrDev** [comment]
I went from mac to linux for work and i miss fork so much, please, make linux happen

**Comment by medzernik** [comment]
+1

**Comment by mannSheth** [comment]
+1

**Comment by jrt324** [comment]
++1

**Comment by gabrieldevsouza** [comment]
+++1
Although I can understand why, I can't believe this is not a thing yet. This software is amazing!

**Comment by ferdbold** [comment]
+1

**Comment by CAinDE** [comment]
+1 :D it's such a great git tool, I miss it!

**Comment by maecky** [comment]
+1

**Comment by eupneo** [comment]
+1

**Comment by SirMartin** [comment]
+1

**Comment by cpetry** [comment]
+1

**Comment by patrick-haldi** [comment]
+1

**Comment by cmcpasserby** [comment]
would love this, but can also 100% see why it won't happen, since at best linux is like 2 extra platforms if a native UI look is required which is one of the big features of fork. Since it would need a GTK and QT version to really pull that off in linux.

**Comment by T4mmi** [comment]
not enterly sure why you would need to maintain both GTK and Qt UIs ? 
Qt is cross-platform and works on both linux/mac/windows so it would actually be a solution to "support them all" :s

**Comment by cmcpasserby** [comment]
@T4mmi was more comment on how what people like about fork is that it uses the native toolkit and conventions for each platform its on, instead of just using electron or something

**Comment by TheZitroX** [comment]
+1

**Comment by GMProg** [comment]
+1

**Comment by luadevsec** [comment]
Someone already tried to run fork.exe (Windows version) with wine or bottles ??

**Comment by rstecca** [comment]
+1

**Comment by rezkhaleghi** [comment]
+1! I'd love to use Fork on Linux. I've been using it on my Mac for a couple of years, and it's been great. It would be awesome to have it available at work too!

**Comment by GhostTheDreamWalker** [comment]
+1

**Comment by escamoteur** [comment]
@DanPristupov you could just use Flutter for the UI and you would cover all platforms. and Dart should be very close to what you are currently using so a port using some AI tool for all the logic part should be pretty straight forward.
Just a thought because not having fork currently holds me from switching to linux

**Comment by serdesize** [comment]
+1

**Comment by Baardi** [comment]
> I've fully moved over to https://github.com/sourcegit-scm/sourcegit source git. Been using Fork for years, but unfortunately, it's a bit jarring switching between 2 git guis. Looks really modern and I actually think it's the best one out there atm.

Sourcegit is chinese made, so for security reasons I'm a bit wary about it. Still use it though, for Linux only. Fork is still better, so I use it where I can

**Comment by rdmrtn** [comment]
> > I've fully moved over to https://github.com/sourcegit-scm/sourcegit source git. Been using Fork for years, but unfortunately, it's a bit jarring switching between 2 git guis. Looks really modern and I actually think it's the best one out there atm.
> 
> Sourcegit is chinese made, so for security reasons I'm a bit wary about it. Still use it though, for Linux only. Fork is still better, so I use it where I can

Do you have Fork's and your operating system's source code?

**Comment by lucastucious** [comment]
> Do you have Fork's and your operating system's source code?

That's not the place to start this kind of debate.

**Comment by philipandresen** [comment]
+1

I will be leaving windows prior to WIN 11 and it seems like a linux operating system is going to be the best choice for me. I will miss fork! I basically learned version control using this app.

**Comment by Danielku15** [comment]
I am also considering to move to a Dual-Boot setup (Linux for Everyday stuff and development, Windows only for gaming where needed). Fork would be one of the main tools I'd be missng. 

Not sure how much platform specific code Fork for Windows has, but moving from WPF to Avalonia could be an easy step towards this. Of course bits like the Git CLI downloading needs to take care of Linux binaries and the installer topic is something to tackle, but from what I know about how Fork for Windows is developed (libraries used and features offered) its hould be feasable. 

If I'd have access to the source I certainly would give it a PoC 😁

**Comment by jirp-upc** [comment]
+1

**Comment by Reyhn3** [comment]
I'm joining the crowd who're waiting hopefully for Fork on Linux (native). I switched over from Windows to Linux last year and the only two things I miss are Fork and Directory Opus.

To be quite honest, I've tried 15-20 different types of git clients for Linux (I just googled around and tried them all), and none comes even close to being as good as Fork is.

This ticket is old, and although I don't have any statistics to back it up, I get the feeling that users migrating to Linux has been increasing exponentially the last one or two years. I'm guessing that pausing the current Fork development to focus on a Linux version would actually help in expanding its user base.

**Comment by andgeno** [comment]
> I'm joining the crowd who're waiting hopefully for Fork on Linux (native). I switched over from Windows to Linux last year and the only two things I miss are Fork and Directory Opus.
> 
> To be quite honest, I've tried 15-20 different types of git clients for Linux (I just googled around and tried them all), and none comes even close to being as good as Fork is.
> 
> This ticket is old, and although I don't have any statistics to back it up, I get the feeling that users migrating to Linux has been increasing exponentially the last one or two years. I'm guessing that pausing the current Fork development to focus on a Linux version would actually help in expanding its user base.

Yes, I see the same happening around me - especially developers, of course, that moved to Linux. I switched to Linux as my daily driver and for all development related tasks about a year ago and can very much relate to only really missing Fork and Directory Opus.

In the meantime, I found an alternative regarding Fork. Have you tried [SourceGit](https://github.com/sourcegit-scm/sourcegit)? I wonder who got inspired from whom as both apps appear so similar in UI/UX. 😁 Anyways, I still wouldn't mind if Fork had official support for Linux. I paid for a Fork license already but would do it again just to further support the development of Fork on Linux as Fork (and Sourcegit) is the only GUI client for git that ticks all my boxes.

**Comment by Reyhn3** [comment]
> In the meantime, I found an alternative regarding Fork. Have you tried [SourceGit](https://github.com/sourcegit-scm/sourcegit)? I wonder who got inspired from whom as both apps appear so similar in UI/UX. 😁 Anyways, I still wouldn't mind if Fork had official support for Linux. I paid for a Fork license already but would do it again just to further support the development of Fork on Linux as Fork (and Sourcegit) is the only GUI client for git that ticks all my boxes.

Thanks for the tip! Yes, it's SourceGit I've been using for the last six months. It's very similar and also very good, but not as polished as Fork. I really like Fork's keyboard scheme, but SourceGit doesn't allow me to change them. And every time SourceGit is updated, the package hash mismatches so Discover (on Fedora) always break and I have to update my whole system manually, which is annoying. Currently looking into [Gittyup](https://flathub.org/apps/com.github.Murmele.Gittyup), which seems like a cross breed between Fork and GitKraken.

**Comment by zekecato** [comment]
+1

I'm moving my primary setup to Debian, with with a Windows machine I can boot up when needed. I'd love to have Fork on Linux as well.

**Comment by andgeno** [comment]
What's the matter with deleting our comments? This actually leaves a bad taste in my mouth and drives me away from your product! If there's no reasonable explanation for deleting the comments I can no longer recommend Fork.

![Image](https://github.com/user-attachments/assets/e474b9ec-1481-49b6-9f80-4e30f5fcefd2)

**Comment by DanPristupov** [comment]
@andgeno This is the Fork's issue tracker. If you want to discuss other git clients, there's probably a better place to do it.
I removed the off-topic.

**Comment by Baardi** [comment]
> What's the matter with deleting our comments? This actually leaves a bad taste in my mouth and drives me away from your product! If there's no reasonable explanation for deleting the comments I can no longer recommend Fork.

I'm one of the guys that got a comment deleted, and I completely support the deletion of comments. He let the important one stay:

> That's not the place to start this kind of debate.

As said by DanPristupov, we went off topic

**Comment by Danielku15** [comment]
@DanPristupov As there is quite a bit of activity on this issue I was wondering if you might have re-discussed this matter on your side? 

My 2 cents: Looking at your tech-stack it might be more feasible to use the Windows (C# / .net) codebase as foundation for a Linux port compared to the Mac version (Swift?). [Avalonia](https://avaloniaui.net/) is an open source WPF alternative sharing a lot of common classes and principles. There are certain differences which require development effort (sometimes drawbacks, sometimes improvements compared to WPF). The biggest efforts typically lie in any custom controls (e.g. merge editor in your case). The more "MVVM" you might follow, the easier replacing the UI layer. 

[Avalonia XPF](https://avaloniaui.net/xpf) could be an option for even lower effort migration but its quite a price investment. If a PoC with a trial license works well (you can develop and test it on Windows using WSL), a crowd funding campaign could finance the license costs of the first few years, giving you time to gradually migrate the required bits.

**Comment by cristian-recoseanu** [comment]
> [@DanPristupov](https://github.com/DanPristupov) As there is quite a bit of activity on this issue I was wondering if you might have re-discussed this matter on your side?
> 
> My 2 cents: Looking at your tech-stack it might be more feasible to use the Windows (C# / .net) codebase as foundation for a Linux port compared to the Mac version (Swift?). [Avalonia](https://avaloniaui.net/) is an open source WPF alternative sharing a lot of common classes and principles. There are certain differences which require development effort (sometimes drawbacks, sometimes improvements compared to WPF). The biggest efforts typically lie in any custom controls (e.g. merge editor in your case). The more "MVVM" you might follow, the easier replacing the UI layer.
> 
> [Avalonia XPF](https://avaloniaui.net/xpf) could be an option for even lower effort migration but its quite a price investment. If a PoC with a trial license works well (you can develop and test it on Windows using WSL), a crowd funding campaign could finance the license costs of the first few years, giving you time to gradually migrate the required bits.

The other major alternative would be something like the [UNO platform](https://platform.uno/platform/).

There seem to be some available resources around migrating WPF:
* [https://platform.uno/wpf/](https://platform.uno/wpf/)
* [https://platform.uno/docs/articles/wpf-migration.html](https://platform.uno/docs/articles/wpf-migration.html)
* [https://github.com/unoplatform/uno/discussions/14548](https://github.com/unoplatform/uno/discussions/14548)

**Comment by CMThF** [comment]
Waiting so hard on this one. I am a long time user & buyer of Fork from the early days on. Moved to linux for work 3 years ago, and I would need this so much :D

**Comment by gamedevsam** [comment]
I have moved back to Windows and do all my work with WSL now because of GitFork, it's that important to my workflow. Would the developers consider doing a fund-raiser to measure demand? I would pre-buy to have this on Linux and / or Windows on ARM (emulated GitFork is noticeably slower than native).

**Comment by ZiBnv** [comment]
+1

**Comment by Nashy1232** [comment]
I too would pay for a Linux version

**Comment by dan-hough** [comment]
I've also previously paid for the Mac/Win version and I would happily pay again for a Linux version.

**Comment by medzernik** [comment]
I'd also gladly pay again. The GUI toolkits are either GTK or QT, QT having more powerful features, but GTK being possibly easier to use (Ghostty also adopted GTK as the main toolkit, and we have Git Fourchette available for QT).

**Comment by Danielku15** [comment]
I lately gained some experience on a hands-on migration of a WPF app to Avalonia and wanted to share some insights. Hope that might help Dan and Tanya to judge the effort and impact to such a transition. I only listed points I think will be relevant for translating Fork. 

I think its feasible for Fork to transition, but it is quite a significant effort (mainly because of custom controls, templates and themeing). 

<details><summary>Open me</summary>
<p>

**General Tips:** 

* Mostly you can just to a bulk rename of your .xamls to .axaml and replace some namespaces. 
* net10.0 "extension members" can help significantly to write "bridges" from old WPF code to Avalonia
	* Beside extension members/methods also use global using aliases 
	* Use "inline this" refactorings for automatic translation of your codebase. 
* The conversion can nicely show you some architectural flaws (UI code pieces where they do not belong)
	* Only refactor anything AFTER the avalonia translation or things get messy. Keep things 'dirty' during the transition phase. 
* A lot of code can be migrated by advanced regex search&replace or using features like [Structural Search and Replace﻿](https://www.jetbrains.com/help/resharper/Navigation_and_Search__Structural_Search_and_Replace.html) if you have R# or Rider.  

**Specific challenges you will face:** 

* All Custom Styles and Themes need to be rewritten for Avalonia
	* Avalonia has no Triggers making this quite a heavyweight effort
	* Some libraries like **Xaml.Behaviors** can take help bridging this gap. 
	* Generally you should use the CSS-alike approach with classes and styles (instead of having a single style loaded from resources) 

* Custom Window Styles and Non-Client area Controls
	* It's possible with Avalonia but needs different implementation to Windows specific WndProc APIs. 
	* Especially the close/minimize/maximize controls you might to rely on the standard implementation. 

* Custom Controls
	* There is no global mouse and keyboard access. You need to forward any details from the event handler args (if available). Global window listeners can help tracking a quasi-global state but you need to lookup the parent window of your current scope. 
	* Avalonia has the concept of "Pointers" (multitouch) instead of a simple mouse. Most mouse logic needs some rework due to alternative APIs. 
	* No OnPreviewEvent and sparse OnEvent methods. You can register to handlers in the constructor instead. 
	* Replace all DependencyProperty with StyledProperties (registration is different, advanced regex replaces can help)
	* There are no "RoutedCommand" and "ApplicationCommand". Which makes some global hooks tricky. These need rework. 
	* Replace any child control lookup of control templates (needs a namescope)
	* Replace any Drag&Drop logic. Especially especially the transfer object has more restrictions. 
	* Replace the animations as Avalonia animations are having a different API
	* Move to other (better) Adorner System -> No special adorner wrapper but you can simply add controls
	* Replace Visibility -> IsVisible
* OS specific APIs need wrappers or alternative approaches (abstraction via DI or simple OS checks during runtime)
	* OS level notifications (toasts)
	* OS integrations like launching apps or asking for system info
	* Credential Manager
	* Dare to drop OS specific behaviors/features (it can indicate bad design decisions, you might think of alternative approaches of serving the related usecase)

* Native dependencies can become a nightmare, consider redevelop the code in .net (if you own it) or look for libraries already serving all platforms you need. For C-style PInvokes you might consider cross-compilation (but serving all architectures can become a cross-compilation hell). 

* Some operations in avalonia are Task-async (e.g. ShowDialog())
* Window.ResizeMode offers less control

* Avalonia has no Freezable. 
	* You could opt for immutable objects if needed or simply drop the freezing logic).

* Replace all ImageSources with IImage 
	* Image loading might be different
	* For drawing images on-the-fly you might use alternative APIs. 
	* Mostly it's  straight forwards search&replace (maybe some extension methods are needed)
	
* Exchange any 3rd party libraries you might use.
* No built-in SpellChecker
	* You might use platform specific APIs or use a 3rd party library
* No built-in HyperLink for textblocks 
	* develop a custom one with InlineUIContainer and HyperlinkButton


Avalonia XPF implements all compatible APIs but I think it's quite an expensive toolkit (30k for 1 app 🫨). A trial might be worth to just evaluate what platform specific problems you have to solve beside the pure WPF UI layer. 

</p>
</details>

**Comment by atqamz** [comment]
will patiently wait for this!

**Comment by aeberbach** [comment]
I would rather pay for Fork all over again to get a Linux version, rather than pay for something else.

**Comment by theburntcrumpet** [comment]
It would be nothing short of a Christmas Miracle, seeing a Linux version of fork

**Comment by Danielku15** [comment]
With an aching heart I have to admit: I finally found a fully cross platform open source alternative (C# with Avalonia, MIT) to Fork I will start using. Fork was one of the final tools that kept me staying on Windows.

If one day Fork ships a Linux version too, I'll consider coming back as I really love what Dan and Tanya are doing and I'd love to continue supporting their work. I've been using Fork since it was a freeware and instantly bought it when it became a shareware. It consumed at least as many CPU time as most of my IDEs and has been my daily companion at work and for private development.

A big thank you for all the work and improvements throughout this year.♥️ 

I'll keep an eye to this issue to stay in the loop if something changes. 

To the community: I don't want to advert the other software as this wouldn't be nice towards Dan and Tanya. Feel free to contact me separately if you need sauce 😉

**Comment by ethanturk** [comment]
Want to throw my hat in the ring for this, too! I'm missing Fork on my linux machine.

The development effort could be mitigated using some nice reverse promptfu ("Given this existing codebase, generate a prompt to build this in Avalonia").

**Comment by epicserve** [comment]
> will patiently wait for this!

It's hard to be patient.

I just posted this Omarchy discussion, and others here might find it relevant.

https://github.com/basecamp/omarchy/discussions/4067

**Comment by noidexe** [comment]
> I would rather pay for Fork all over again to get a Linux version, rather than pay for something else.

Me too. If the money is needed in advance I'd be happy to support a kickstarter campaign or use any other preorder method.

**Comment by CedricPerzik** [comment]
Came to say I never bought fork and used the free version on windows on and off, but i have now come to a point where i left windows behind and your application is the thing I miss most. Though I didn't purchase back then, I would happily throw my entire wallet at you to get a Linux version.

**Comment by Magehernan** [comment]
Wow it started on 2017, me too migrate to linux trying to scape microslop, and i use Fork for years for free, and the like a year ago i purchased it, now on linux i am trying to make it my main system so i was so down when i realize that fork don't work on linux :( i am testing using winboat to use visual studio 2026 so i install fork on it, i hope some day we have fork on linux and i understand because all the GUI implementations there are it will a task. Thanks for all the work en fork it is great.

**Comment by FurkanEmiroglu** [comment]
Linux really deserves a nice GUI tool like Fork. Such a shame.

**Comment by EdMueller** [comment]
+1 for Fork on Linux. Please ;-)

**Comment by rubickanov** [comment]
+1

**Comment by epicserve** [comment]
Until there is a Linux version of Fork, [SourceGit](https://github.com/sourcegit-scm/sourcegit) is the closest thing I've found.

**Comment by flames5123** [comment]
More and more people are moving to Linux given Windows 11's aggressive AI implementation. Now is the time for fork to have a Linux version. Even game devs don't have to do anything special for Linux, so it could be super simple for you to implement.

**Comment by FearlessHyena** [comment]
This would be great to have! I use the paid version of Fork for both my Mac and Windows environment but like a lot of people fed up with Microslop I'm moving to Linux too 😺
I think getting it to work with Proton/Wine first would be great. A native app might take time so is fine as a future project

**Comment by silvio-vallorani** [comment]
+1 for Fork on Linux! 
Pleeeeease!

**Comment by FractalCodeRicardo** [comment]
Linux version would be awesome

**Comment by aZavarella** [comment]
Should a native library eventually become good enough, I hope there will be news on this topic.

**Comment by NicolasAubinet** [comment]
Just switch to Linux as well, went to download Fork and just realized it doesn't exist on Linux. Been using Fork for years, hopefully it comes to Linux some day. Will need to find an alternative in the meantime.

**Comment by staus** [comment]
Also adding to the voices. Would like to pay 3 times for a Linux version. Good work like Fork is hard to come by.

**Comment by Nashy1232** [comment]
The fact they delete any comment regarding any alternatives is really stupid... especially when we all get emailed the original comment anyway.

Honestly, The lack of Linux support has pushed me away and I've found an alternative I like that works on both windows and Linux & I personally think its better. So, I doubt I'm ever going back to fork.

**Comment by ivpeykov** [comment]
Your app made switching to linux a hard decision. Would pay good money for a port.

**Comment by C0untZero** [comment]
+1 for Fork on linux!

**Comment by Tornith** [comment]
+1 for a Linux port, wanting to make the jump to Linux, but losing Fork terrifies me 😭

**Comment by SamDiaz9** [comment]
One here for the linux version! The only program im really mising after leaving windows, it will be great having this as a possibility :)

**Comment by tanj** [comment]
+1 for linux port. I can do everything via command-line, but fork makes everything so easy and lays out so well. I miss fork when I working in linux.

**Comment by rubickanov** [comment]
+1

**Comment by espen** [comment]
There is no need to add +1 comments, it only spams email of those subscribing to this issue. Please upvote using the emoji in the original issue comment.

**Comment by DanPristupov** [comment]
> There is no need to add +1 comments, it only spams email of those subscribing to this issue. Please upvote using the emoji in the original issue comment.

Yes, I agree 👍. If anyone wants to support this, please upvote the original https://github.com/fork-dev/Tracker/issues/153#issue-271372916.

**Comment by pfeerick** [comment]
And if you want to get updates, there is no need to post a message either, there is a subscribe option in the right sidebar.

**Comment by cristian-recoseanu** [comment]
I bought Fork and use it on both Windows and MacOS. It is indispensable, however after migrating on my personal machine to Fedora it is sorely missed.
In the meantime I am making do with:
https://github.com/sourcegit-scm/sourcegit
